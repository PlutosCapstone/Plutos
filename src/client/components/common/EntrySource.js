import React from "react";
import { StyleSheet, Text, View } from "react-native";
import RightArrow from "../../assets/icons/RightArrow";
import GroupIcon from "../../assets/icons/GroupIcon";

const EntrySource = ({ incomeTitle, additionalInfo, group }) => {
  return (
    <View style={styles.box}>
      <View style={styles.textBox}>
        {group ? (
          <View style={styles.group}>
            <Text style={[styles.text]}>{incomeTitle}</Text>
            <View style={styles.groupIcon}>
              <GroupIcon size={30} />
            </View>
          </View>
        ) : (
          <Text style={[styles.text, { width: "65%" }]}>{incomeTitle}</Text>
        )}
        <Text style={[styles.text, { width: "35%", textAlign: "center" }]}>
          {additionalInfo}
        </Text>
      </View>
      <View style={styles.rightArrow}>
        <RightArrow />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  box: {
    display: "flex",
    flexDirection: "row",
    position: "relative",
    marginVertical: 8,
  },
  textBox: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    width: "85%",
    height: 50,
  },
  rightArrow: {
    position: "absolute",
    right: 0,
  },
  text: {
    fontSize: 18,
  },
  group: {
    width: "65%",
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
  },
  groupIcon: { marginLeft: 8 },
});

export default EntrySource;
