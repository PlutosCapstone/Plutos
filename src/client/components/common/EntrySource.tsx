import React from "react";
import { StyleSheet, Text, View } from "react-native";

interface EntrySourceProps {
  description: string;
  additionalInfo: string;
}

const EntrySource = ({ description, additionalInfo }: EntrySourceProps) => {
  return (
    <View style={styles.box}>
      <View style={styles.textBox}>
        <Text style={[styles.text, { width: "65%" }]}>{description}</Text>
        <Text style={[styles.text, { width: "35%", textAlign: "center" }]}>
          {additionalInfo}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  box: {
    display: "flex",
    flexDirection: "row",
    position: "relative",
    marginVertical: 8,
  },
  textBox: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    width: "100%",
    height: 50,
    marginLeft: 5,
  },
  text: {
    fontSize: 18,
  },
});

export default EntrySource;
