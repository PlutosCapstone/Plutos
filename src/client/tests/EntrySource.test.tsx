import { render } from "@testing-library/react-native";
import EntrySource from "../components/common/EntrySource";

describe("EntrySource component", () => {
  it("should match snapshot for entry source component", () => {
    const page = render(
      <EntrySource
        description="description test"
        additionalInfo="additionalInfo test"
      />,
    );

    const { getByText } = page;
    const descriptionText = getByText("description test");
    const additionalInfoText = getByText("additionalInfo test");

    expect(descriptionText).toBeTruthy();
    expect(additionalInfoText).toBeTruthy();

    const tree = page.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
