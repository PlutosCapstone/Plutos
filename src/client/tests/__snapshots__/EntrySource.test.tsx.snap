// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntrySource component should match snapshot for entry source component 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "marginVertical": 8,
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "marginLeft": 5,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 18,
          },
          {
            "width": "65%",
          },
        ]
      }
    >
      description test
    </Text>
    <Text
      style={
        [
          {
            "fontSize": 18,
          },
          {
            "textAlign": "center",
            "width": "35%",
          },
        ]
      }
    >
      additionalInfo test
    </Text>
  </View>
</View>
`;
